### DISCUSSION TITLE
may extract text like in Microsoft Edge

### DISCUSSION THREAD

**Discussion by Eyderoe** [jsvine/pdfplumber#1077] [Ask for help with specific PDFs]
extract table in a pdf.
one row like this:
![屏幕截图 2024-01-08 185924](https://github.com/jsvine/pdfplumber/assets/103725178/c7102acd-a133-48da-8912-953bb2247044)
and run the code
```
pdf = pdfplumber.open(raw)
first = pdf.pages[0]  
texts = first.extract_text_lines()
chars = [i["chars"] for i in texts]
for iChar in chars:
    for info in iChar:
        print("({},{})".format(info["text"], round(info["x0"], 1)), end=' ')
    print('\n------------')
```
and get this
![image](https://github.com/jsvine/pdfplumber/assets/103725178/5edf88b9-ea1f-4461-ab12-c245479dba0c)
magiclly look like extract from 
JC401 N38°31'08.9"E102°35'45G.2" JC N38°33.4'E102°19.4' 
and when I use edge 
1. open pdf in edge
2. ctrl+a ctrl+c
3. open a txt file
4. ctrl+v

I can get this perfect
![image](https://github.com/jsvine/pdfplumber/assets/103725178/efb39de7-7c90-4412-90f7-d0ea95d8c05b)
and edge wont add some strange blank in it (due to some reason I only can show this row)
so, I hope you might add the pdf engine of edge to pdfplumber sincerely
[](url)https://github.com/chromium/chromium/tree/main/pdf
- Sincerely Eyderoe

**Comment by jsvine** [comment]
Hi @Eyderoe, and thanks for your interest in this library. Unfortunately, without more details — e.g., the code you're running — it's difficult for me to understand the particular issue you're having. If the issue is the appearance of extra newline characters produced by methods such as `page.extract_text(...)`, you might try adjusting the `y_tolerance` parameter.

**Reply by Eyderoe** [reply]
I re-edit the whole question now
@jsvine thanks for reply. simply say, I meet a pdf process wrong by pdfplumber and perfect by Egde. I dont know whether its a bug in pdfplumber or Edge more smart. If its a bug, hope might fix it, if Edge more smart hope might add Edge's pdf engine to pdfplumber (edge based on open source chromium.) If I can send file privately, I can order more details include file and so on [please forgive my bad English]

**Reply by jsvine** [reply]
Hi @Eyderoe, and thanks for the additional details. PDFs come in many layouts, structures, et cetera — `pdfplumber` does not aim to extract every single one perfectly with the default settings, but rather provides various settings and parameters to allow users to adjust the output. In this case, Edge may indeed provide a better result for you by default, but its PDF engine is a different type of technology than `pdfplumber` and I don't foresee incorporating it int this project.

**Reply by Eyderoe** [reply]
Thanks, I may try OCR instead
